File: LaptopShop_Backend/LaptopShop_Final5/src/main/java/com/example/demo/LService/CartService.java
package com.example.demo.LService;

import java.util.List;

import com.example.demo.LEntity.Cart;
import com.example.demo.LEntity.Laptop1;

	public interface CartService {
	
		   Cart saveCart(Cart cart);
		    List<Cart> getAllCarts();
		    Cart getCartById(Long id);
		    Cart updateCart(Long id, Cart cart);
		    Cart deleteCart(Long id);
	}

File: FrontEndLaptop/FrontEndLaptop/src/app/cart.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CartService {
  private apiUrl = 'http://localhost:8080/api/carts'; // Update with your Spring Boot API URL

  constructor(private http: HttpClient) {}

  saveCart(cart: any): Observable<any> {
    return this.http.post<any>(`${this.apiUrl}`, cart);
  }

  getAllCarts(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}`);
  }

  getCartById(id: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/${id}`);
  }

  updateCart(id: number, cart: any): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/${id}`, cart);
  }

  deleteCart(id: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${id}`);
  }

  // Other methods for cart-related operations as needed
}
